backend/data.go

```diff
@@ -8,35 +8,55 @@ import (
 	"github.com/grafana/grafana-plugin-sdk-go/data"
 )
 
+// QueryDataHandler handles data queries.
+// req contains the queries []DataQuery (where each query contains RefID as a unique identifer).
+// The QueryDataResponse contains a map of RefID to the response for each query.
+type QueryDataHandler interface {
+	QueryData(ctx context.Context, req *QueryDataRequest) (*QueryDataResponse, error)
+}
+
+// QueryDataRequest contains a single request which contains multiple queries.
+// It is the input type for a QueryData call.
 type QueryDataRequest struct {
 	PluginConfig PluginConfig
 	Headers      map[string]string
 	Queries      []DataQuery
-	User         *User
+	User         *User // User is information about the grafana-server user that made the request.
 }
 
-// DataQuery represents the query as sent from the frontend.
+// DataQuery represents a single query as sent from the frontend.
+// A slice of DataQuery makes up the Queries property of a QueryDataRequest.
 type DataQuery struct {
-	RefID         string
-	MaxDataPoints int64
-	Interval      time.Duration
-	TimeRange     TimeRange
-	JSON          json.RawMessage
+	RefID         string          // RefID is the unique identifer of the query, set by the frontend call.
+	MaxDataPoints int64           // MaxDataPoints is the maximum number of datapoints that should be returned from a time series query.
+	Interval      time.Duration   // Interval is the suggested duration between time points in a time series query.
+	TimeRange     TimeRange       // TimeRange is the Start and End of the query as sent by the frontend.
+	JSON          json.RawMessage // JSON is the raw JSON query and includes the above properties as well as custom properties.
 }
 
-// QueryDataResponse holds the results for a given query.
+// QueryDataResponse contains the results from a QueryDataRequest.
+// It is the return type of a QueryData call.
 type QueryDataResponse struct {
-	Frames   []*data.Frame
-	Metadata map[string]string
+	Responses map[string]*DataResponse // Responses is a map of RefIDs (Unique Query ID) to *DataResponse.
+	Metadata  map[string]string
```

Comment:
remove this?  no clear place for it to go yet?

Reply:
updated with comments

Reply:
oh wait I think I may have mixed this up :-/

Reply:
Errr... I was confused ... removing.

Reply:
Okay, so now: No meta on QueryDataResponse

Meta on DataResponse now has a warning comment.